extension/deployment/src/main/java/io/quarkiverse/temporal/deployment/TemporalProcessor.java

```diff
@@ -157,19 +152,17 @@ void produceActivities(
             BuildProducer<ActivityImplBuildItem> producer) {
 
         Map<DotName, Set<String>> explicitBinding = new HashMap<>();
-        temporalBuildtimeConfig.worker().forEach((worker, config) -> {
-            config.activityClasses().ifPresent(classes -> {
-                for (String activityClass : classes) {
-                    DotName className = DotName.createSimple(activityClass);
-                    ClassInfo classInfo = beanArchiveBuildItem.getIndex().getClassByName(className);
-                    if (doesNotImplementAnnotatedInterface(beanArchiveBuildItem, classInfo, ACTIVITY_INTERFACE)) {
-                        throw new ConfigurationException("Class " + activityClass + " is not an activity");
-                    }
-                    explicitBinding.computeIfAbsent(DotName.createSimple(activityClass), (k) -> new HashSet<>())
-                            .add(worker);
+        temporalBuildtimeConfig.worker().forEach((worker, config) -> config.activityClasses().ifPresent(classes -> {
+            for (String activityClass : classes) {
+                DotName className = DotName.createSimple(activityClass);
+                ClassInfo classInfo = beanArchiveBuildItem.getIndex().getClassByName(className);
+                if (doesNotImplementAnnotatedInterface(beanArchiveBuildItem, classInfo, ACTIVITY_INTERFACE)) {
+                    throw new ConfigurationException("Class " + activityClass + " is not an activity");
                 }
-            });
-        });
+                explicitBinding.computeIfAbsent(DotName.createSimple(activityClass), (k) -> new HashSet<>())
+                        .add(worker);
+            }
+        }));
```

Comment:
same, Ideally if there is no changes I would revert this for the sake of the clarity of the PR if anyone try to go back to this change later